File: leetCode/easy/14_Longest_Common_prefix.js
var longestCommonPrefix = function(strs) {
    for (let i = 0; i<strs[0].length; i++){
        for (let str of strs){
            let firstChar = str[i];
            let curChar = strs[0][i]
            if (firstChar !== curChar){
                return str.slice(0,i)
            }
        }
    }
    return strs[0]
};


// better solution
let longestCommonPrefix2 = function(strs) {
    let prefix = ""
    for(let i=0; i< strs[0].length; i++){
        const character = strs[0][i]
        for(let j = 1;j < strs.length; j++){
            if(strs[j][i] !== character) return prefix
        }
        prefix += character
    }
    return prefix
 };
 longestCommonPrefix(["flower","flow","flight"])

File: leetCode/easy/PY/14_Longest_Common_prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
      if str == []: return ""
      shortest = min(strs, key=len)
      prefix = ""
      for i, char in enumerate(shortest):
            for words in strs:
                if words[i] != char: 
                    return shortest[:i]
      return shortest